### DISCUSSION TITLE
Increase upload limit to be use with Nextcloud

### DISCUSSION THREAD

**Discussion by kbarbora** [evertramos/nginx-proxy-automation#309] [Q&A] [answered]
I had tried to increase the upload file size without success.
client_max_body_size parameter seems useless
Does anyone were able to fix this issue?

I believe has to be modified in the proxy, since I modified the config in nextcloud container and looks like the limits are good.

**Comment by evertramos** [comment, marked as answer]
Hi @kbarbora 

In order to update the nginx option `client_max_body_size` you must do the following:

1. [Update the limit at the `./conf.d/uploadsize.conf`](https://github.com/evertramos/nginx-proxy-automation/blob/9ce0320fff25360dbd7ec2feb61ac1e725f81446/conf.d/uploadsize.conf#L5)

2. Start the proxy with the option `--use-nginx-conf-files`

**Comment by kbarbora** [comment]
Thanks, I actually modified the nginx.tmpl and added `client_max_body_size=20G` in server, location and main blocks and worked. However, I see that the upload speed of nextcloud is limited to 10mb/s, really slow for a SSD setup. Some people suggested to disable HTTP/2 in nginx, any thoughts?

**Reply by evertramos** [reply]
The _nginx.tmpl_ should not be edited directly, as my suggestion, once we always use the updated version of [nginx-proxy](https://github.com/nginx-proxy/nginx-proxy), and about the http/2 you must check in [nginx-proxy](https://github.com/nginx-proxy/nginx-proxy) as well.
